pkg/network/dhcp/server/server_test.go

```diff
@@ -155,49 +157,29 @@ var _ = Describe("DHCP Server", func() {
 			return b
 		}
 
-		It("should reject domains that start with '-'", func() {
-			dom := "-foo.com"
-			Expect(isValidSearchDomain(dom)).To(BeFalse())
-		})
-
-		It("should reject domains with full length greater than 253 chars", func() {
-			b := append(createBytes(250), []byte(".com")...)
-			Expect(isValidSearchDomain(string(b))).To(BeFalse())
-		})
-
-		It("should accept domains with full length of 253 chars", func() {
+		DescribeTable("should *REJECT* domains", func(domaintToTest string) {
+			Expect(isValidSearchDomain(domaintToTest)).To(BeFalse())
+		},
+			Entry("that start with '-'", "-foo.com"),
+			Entry("that end with '-'", "foo.com-"),
+			Entry("with full length greater than 253 chars", string(append(createBytes(250), []byte(".com")...))),
+			Entry("that have labels longer than 63 chars", string(append(createBytes(64), []byte(".com")...))),
+			Entry("with invalid characters", "foo\n.com"),
+		)
+
+		DescribeTable("should *ACCEPT* domains", func(domaintToTest string) {
+			Expect(isValidSearchDomain(domaintToTest)).To(BeTrue())
+		},
+			Entry("that have 63 character labels", string(append(createBytes(63), []byte(".com")...))),
+			Entry("with a valid domain name", "example.default.svc.cluster.local"),
+			Entry("with a valid FQDN", "example.default.svc.cluster.local."),
+			Entry("with a partial search domain", "local"),
+		)
+
+		XIt("should accept domains with full length of 253 chars", func() {
```

Comment:
Can we please either fix the test or remove it? `XIt` should not be used.

Reply:
The whole point is I don't have time right now to fix the test. I agree that would be the best course of action. 

Deleting it, is not an option, since that would be just masking an issue; the test shows that there is something clearly wrong on our code base. 

I'm attempting to take an intermediate route by marking the test as broken.  (OK, `X` means pending ...). Would it be OK if I use this `XFail` [construct](https://github.com/kubevirt/kubevirt/blob/099be903803556bfe3a85075a7b55f0a711d9ca7/tests/assert/xfail.go) ? afaiu, this exists for this kind of stuff.

Do you really prefer the test to be left broken as it currently is ? I.e. "should accept ..." description, then have an expectation like `Expect(accepts(...)).To(BeFalse())`

Reply:
Never mind, I thought for a moment that the `XIt` is similar to `XFail`. `XIt` or `PIt` is fine. `XFail` should not be used.